### PR TITLE
load_csv/feat: support multi csv field

### PR DESCRIPTION
**Pull request by avelino**
fixed: #146

### CODE REVIEW

File: src/load_csv.py [hunk 1]
```diff
@@ -67,6 +68,8 @@ def load_csv_and_generate_embeddings(path, cleardb=False):
     parser = argparse.ArgumentParser()
     parser.add_argument("--path", type=str, required=False, default="./know.csv")
     parser.add_argument("--cleardb", action="store_true")
+    parser.add_argument("--columns", type=str, required=False, default="content")
```
Comment (walison17) [inline src/load_csv.py:71]:
```suggestion
    parser.add_argument("--columns", nargs="*", default=["content"])
```

File: src/load_csv.py [hunk 2]
```diff
@@ -54,7 +54,8 @@ def load_csv_and_generate_embeddings(path, cleardb=False):
     print("New questions:", len(df_filtered))
 
     df_filtered.drop(columns=["primary_key"], inplace=True)
-    df_filtered["embedding"] = generate_embeddings(df_filtered["content"])
+    df_filtered["embedding"] = generate_embeddings(
+        "\n".join([df_filtered[column] for column in columns]))
```
Comment (walison17) [inline src/load_csv.py:58]:
```suggestion
        "\n".join(df_filtered[column] for column in columns))
```

File: src/load_csv.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 from dialog.models.db import session
 
 
-def load_csv_and_generate_embeddings(path, cleardb=False):
+def load_csv_and_generate_embeddings(path, cleardb=False, columns=["content"]):
```
Comment (walison17) [inline src/load_csv.py:12]:
```suggestion
def load_csv_and_generate_embeddings(path, cleardb=False, columns=("content",)):
```

**Comment by lgabs** [comment]
Actually this part needs to be changed to support this possibility of embedding multiple columns at once (typical case for "question+answer" in the same column (aka a document). 

https://github.com/talkdai/dialog/blob/f97728a06883d3deeb1e1af58006be0fd92b56ef/src/dialog/llm/default.py#L41

File: src/load_csv.py [hunk 4]
```diff
@@ -67,6 +68,8 @@ def load_csv_and_generate_embeddings(path, cleardb=False):
     parser = argparse.ArgumentParser()
     parser.add_argument("--path", type=str, required=False, default="./know.csv")
     parser.add_argument("--cleardb", action="store_true")
+    parser.add_argument("--columns", type=str, required=False, default="content")
     args = parser.parse_args()
 
-    load_csv_and_generate_embeddings(args.path, args.cleardb)
+    load_csv_and_generate_embeddings(
+        args.path, args.cleardb, args.columns.split(","))
```
Comment (walison17) [inline src/load_csv.py:75]:
```suggestion
        args.path, args.cleardb, args.columns)
```

**Comment by avelino** [comment]
> Actually this part needs to be changed to support this possibility of embedding multiple columns at once (typical case for "question+answer" in the same column (aka a document).
> 
> https://github.com/talkdai/dialog/blob/f97728a06883d3deeb1e1af58006be0fd92b56ef/src/dialog/llm/default.py#L41

@lgabs the goal of this PR is to generate embeddings with more fields, and not to change the way, we work with prompt generation for OpenAI.

> We can discuss this matter in an issue and understand why this behavior should be changed, but this is not the place for that discussion.

**Comment by lgabs** [comment]
> > Actually this part needs to be changed to support this possibility of embedding multiple columns at once (typical case for "question+answer" in the same column (aka a document).
> > https://github.com/talkdai/dialog/blob/f97728a06883d3deeb1e1af58006be0fd92b56ef/src/dialog/llm/default.py#L41
> 
> @lgabs the goal of this PR is to generate embeddings with more fields, and not to change the way, we work with prompt generation for OpenAI.
> 
> > We can discuss this matter in an issue and understand why this behavior should be changed, but this is not the place for that discussion.

I've said that because we'll probably create confusion: when we allow the embedding of multiple fields, the typical motivation case will be question+answer together, which usually gives much better results in RAG from user queries, and keeping this behavior in `default.py` will make the prompt generation duplicates the question. We could avoid this confusion explaining that `content` so far means _"answer to a question"_ and later discuss towards a more concise solution.

**Comment by vmesel** [comment]
> we'll probably create confusion: when we allow the embedding of multiple fields, the typical motivation case will be question+answer together

This is still served as multiple-fields embedding, so we do not need to address to a specific solution, if a user decides the question is not relevant to the context, he should be able to change it easily without editing our source code.

> We could avoid this confusion explaining that content so far means "answer to a question" and later discuss towards a more concise solution.

The user can set this into the prompt settings.

File: src/load_csv.py (hunk 3 above)
Comment (vmesel) [reply src/load_csv.py]: @walison17 why are we fallbacking to a tuple here?

**Comment by vmesel** [comment]
@avelino @walison17 LGTM, but i'd rather have the columns tuple as a list instead of a tuple.

**Review by vmesel — COMMENTED** [review]
LGTM!

**Review by vmesel — APPROVED** [review]

**Pull request merged**